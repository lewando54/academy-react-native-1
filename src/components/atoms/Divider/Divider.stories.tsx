/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react'

import Divider from './Divider'
import { create } from 'react-test-renderer'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
  component: Divider,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    text: { control: 'text' }
  }
} satisfies Meta<typeof Divider>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
export default meta
type Story = StoryObj<typeof meta>

export const WithText: Story = {
    args: {
        text: 'Divider text',
        testId: 'story'
    }
}

export const WithoutText: Story = {
    args: {
        text: '',
        testId: 'story'
    }
}
// More on args: https://storybook.js.org/docs/react/writing-stories/args
