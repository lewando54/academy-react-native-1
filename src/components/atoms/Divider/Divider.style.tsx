import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    div:{
        display: 'flex',
        width: '100%',
        alignItems: 'center',
        flexDirection: 'row'
    },
    hr:{
        flex: 1,
        height: 2,
        backgroundColor: '#e5e7eb'
    },
    text:{
        padding: 4,
        borderWidth: 2, 
        borderColor: '#e5e7eb',
        borderRadius: 6,
        fontSize: 14,
        lineHeight: 20,
        margin: 0,
        textAlign: 'center',
        color: 'rgb(156 163 175)'
    }
})