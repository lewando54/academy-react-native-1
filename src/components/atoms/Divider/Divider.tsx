import React from 'react'
import { Text, View } from 'react-native'
import dividerStyle from './Divider.style'

interface IDividerProps {
    text?: string
    testId: string
}

function Checkbox(props: IDividerProps): React.ReactElement{
    return (
            <View style={dividerStyle.div}>
                <View style={dividerStyle.hr}></View>
                {props.text && <Text style={dividerStyle.text}>{props.text}</Text>}
                <View style={dividerStyle.hr}></View>
            </View>
    )
}

export default Checkbox