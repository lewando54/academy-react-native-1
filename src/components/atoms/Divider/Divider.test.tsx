import React from 'react'
import * as renderer from 'react-test-renderer'
import Divider from './Divider'

describe('Divider', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Divider testId='divider'/>)
    expect(tree).toMatchSnapshot()
  })

  it('should render with optional text', () => {
    const tree = renderer.create(<Divider testId='divider' text='Hello World!' />)
    expect(tree).toMatchSnapshot()
  })

  it('should not render text by default', () => {
    const tree = renderer.create(<Divider testId='divider' />)
    expect(tree).toMatchSnapshot()
  })
})
