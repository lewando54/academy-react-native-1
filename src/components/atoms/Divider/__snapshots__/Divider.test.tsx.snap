// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should not render text by default 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#e5e7eb",
        "flex": 1,
        "height": 2,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#e5e7eb",
        "flex": 1,
        "height": 2,
      }
    }
  />
</View>
`;

exports[`Divider should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#e5e7eb",
        "flex": 1,
        "height": 2,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#e5e7eb",
        "flex": 1,
        "height": 2,
      }
    }
  />
</View>
`;

exports[`Divider should render with optional text 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#e5e7eb",
        "flex": 1,
        "height": 2,
      }
    }
  />
  <Text
    style={
      {
        "borderColor": "#e5e7eb",
        "borderRadius": 6,
        "borderWidth": 2,
        "color": "rgb(156 163 175)",
        "fontSize": 14,
        "lineHeight": 20,
        "margin": 0,
        "padding": 4,
        "textAlign": "center",
      }
    }
  >
    Hello World!
  </Text>
  <View
    style={
      {
        "backgroundColor": "#e5e7eb",
        "flex": 1,
        "height": 2,
      }
    }
  />
</View>
`;
